### ISSUE TITLE
The SqlDbType enumeration value, 240, is invalid.

### ISSUE THREAD

**Issue by malekpour** [dotnet/efcore#2555]
When current profile is dnxcore50 calling SqlDataReader.NextResult or NextResultAsync throws exception with following message:

```
{"The SqlDbType enumeration value, 240, is invalid.\r\nParameter name: SqlDbType"}
```

Stack trace:

```
   at System.Data.SqlClient.MetaType.GetSqlDataType(Int32 tdsType, UInt32 userType, Int32 length)
   at System.Data.SqlClient.TdsParser.TryCommonProcessMetaData(TdsParserStateObject stateObj, _SqlMetaData col)
   at System.Data.SqlClient.TdsParser.TryProcessMetaData(Int32 cColumns, TdsParserStateObject stateObj, _SqlMetaDataSet& metaData)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.TryNextResult(Boolean& more)
   at System.Data.SqlClient.SqlDataReader.<>c__DisplayClass0.<NextResultAsync>b__1(Task t)
   at System.Data.SqlClient.SqlDataReader.InvokeRetryable[T](Func`2 moreFunc, TaskCompletionSource`1 source, IDisposable objectToDispose)
```

**Comment by rowanmiller** [comment]
Hey,

Are you hitting this with EF7? Can you provide some code we can use to reproduce the issue?

~Rowan

**Comment by malekpour** [comment]
Hi, Not EF7, just pure SqlDataReader. Here is the code...

```csharp
public async Task Main(string[] args)
{
    var constr = @"Data Source=.\SqlExpress;Initial Catalog=AdventureWorks;Integrated Security=True";

    var query = @"
SELECT * FROM [AdventureWorks].[Person].[Person]
SELECT * FROM [AdventureWorks].[Person].[Address]
";
    using (var connection = new SqlConnection(constr))
    using (var command = new SqlCommand(query, connection))
    {
        command.Connection.Open();

        var reader = await command.ExecuteReaderAsync();

        while (await reader.ReadAsync())
        {
        }

        await reader.NextResultAsync(); // Error happens here only on CoreClr

        while (await reader.ReadAsync())
        {
        }
    }
}
```
```json
  "frameworks": {
    "dnx451": {
      "frameworkAssemblies": {
        "System.Data": ""
      }    
    },
    "dnxcore50": {
      "dependencies": {
        "System.Data.SqlClient": "4.0.0-*",
        "System.Console": "4.0.0-*",
        "System.Collections": "4.0.10-*",
        "System.Linq": "4.0.0-*",
        "System.Threading": "4.0.10-*",
        "Microsoft.CSharp": "4.0.0-*"
      }
    }
  }
```

**Comment by malekpour** [comment]
Also this may help since it happens only after second ```SELECT *``` statements.

```csharp
public void Main(string[] args)
{
    var constr = @"Data Source=.\SqlExpress;Initial Catalog=AdventureWorks;Integrated Security=True";

    var query = @"
SELECT * FROM [AdventureWorks].[Person].[Person]                -- okay
SELECT [ModifiedDate] FROM [AdventureWorks].[Person].[Address]  -- okay
SELECT * FROM [AdventureWorks].[Person].[Address]               -- error
";
    using (var connection = new SqlConnection(constr))
    using (var command = new SqlCommand(query, connection))
    {
        command.Connection.Open();

        var reader = command.ExecuteReader();

        while (reader.Read())
        {
        }

        reader.NextResult();

        while (reader.Read())
        {
        }

        reader.NextResult();

        while (reader.Read())
        {
        }
    }
}
```

**Comment by rowanmiller** [comment]
@malekpour I've contacted the SQL Client team to take a look at this one. They aren't on GitHub, so I'll leave this issue open until we have confirmation from them that the issue is being tracked elsewhere.

**Comment by dazinator** [comment]
Is there any update to this issue?

I am using EF7 and have started randomly getting a similar error: only it says it's type 0 is not valid:

```
![image](https://cloud.githubusercontent.com/assets/3176632/13253924/a901e676-da36-11e5-8bb1-9881a0fa66fc.png)

```

**Comment by rowanmiller** [comment]
@dazinator the original issue from this bug was fixed a while back. Could you open a new issue for the SQL Client team on https://github.com/dotnet/corefx.

**Comment by dazinator** [comment]
@rowanmiller - I fixed a bug today in our implementation, where we wern't managing the liftetime of `DbContext` correctly - i.e some where sitting the `IServiceProvider` at the application services level rather than `RequestServices` and so were not being disposed, and were sometimes being re-used accorss different requests. I have a feeling that this may not have been helping matters. Now that this is fixed, if I still encounter the above error then I will be sure to raise an issue, otherwise I will assume this was the cause ;)

**Comment by rowanmiller** [comment]
@dazinator :+1:

**Comment by tbluesky2012** [comment]
I have the same error, and when I add 'MultipleActiveResultSets=true' in my connection string, the error is disappeared.

**Comment by tbluesky2012** [comment]
I had the same error, and when I add 'MultipleActiveResultSets=true' in my connection string, the error was disappeared.

**Comment by roji** [comment]
@tbluesky2012 that sounds like a SqlClient bug, can you open an issue with a code sample on https://github.com/dotnet/SqlClient?
